client/azure.go

```diff
@@ -194,6 +209,20 @@ func newVirtualNetworkClient(config *azureClientConfig) (*network.VirtualNetwork
 	return &client, nil
 }
 
+func newVirtualMachineScaleSetsClient(config *azureClientConfig) (*compute.VirtualMachineScaleSetsClient, error) {
+	client := compute.NewVirtualMachineScaleSetsClient(config.subscriptionID)
+	client.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
+
+	return &client, nil
+}
+
+func newVirtualMachineScaleSetVMsClient(config *azureClientConfig) (*compute.VirtualMachineScaleSetVMsClient, error) {
+	client := compute.NewVirtualMachineScaleSetVMsClient(config.subscriptionID)
+	client.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
+
+	return &client, nil
```

Comment:
When there is no need to return `error` it should be dropped from the function signature. I got that this is aligned with other `new*Client` func, but this should be refactored IMO. Maybe in a different PR though.

Reply:
Tracking in https://github.com/giantswarm/giantswarm/issues/3355.